services/manage.ps1

```diff
@@ -0,0 +1,223 @@
+    [string] $name = [System.Guid]::NewGuid()
+    New-Item -ItemType Directory -Path (Join-Path $parent $name)
+}
+
+function Invoke-Build {
+    param ([switch]$All, [switch]$NoCache)
+
+    $cache = ""
+    if ($NoCache) {
+        Write-Host ">>> Building $service without cache ..."
+        $cache = "--no-cache "
+    }
+
+    if ($All) {
+        foreach ($service in $services) {
+            Write-Host ">>> Building $service ..."
+            & docker build -t "ynput/ayon-shotgrid-$($service):$($addon_version)" -f "$($repo_root)\services\$($service)\Dockerfile" $repo_root\services
+        }
+    } else {
+        Write-Host ">>> Building $service ..."
+        & docker build --network=host -t $image -f $repo_root\services\$service\Dockerfile $repo_root\services
+    }
+    Write-Host ">>> Done."
+}
+
+
+function Clear-Images {
+    Write-Host ">>> Looking for $image ..."
+    if ($null -ne (docker images -a --format '{{print .Tag | print ":" | print .Repository}}' | Select-String -Pattern $image)) {
+        Write-Host ">>> Removing $image ..."
+        & docker rmi $image
+    }    
+}
+
+function Start-Dev {
+    Write-Host ">>> Starting $service ..."
+
+    <#
+
+    $IsAdministrator = ([Security.Principal.WindowsPrincipal]::new(
+        ($id = [Security.Principal.WindowsIdentity]::GetCurrent())
+    )).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
+    $id.dispose(); remove-variable id
+
+    if (-not $IsAdministrator) {
+        Write-Host "!!! Warning: This must be run from a PowerShell terminal with Administrator privileges !!!" -ForegroundColor Yellow
+        $confirmation = Read-Host "Running without them might not work. Are you sure you want to proceed [y/n]: "
+        if ($confirmation -eq 'n') {
+            Write-Host ">>> Aborting."
+            Exit-WithCode 1
+        }
+    }
+
+    Write-Host ">>> Creating symbolic link for shotgrid_common ..."
+    $dir_path = "$($repo_root)\services\shotgrid_common"
+    $target_dir = "$($repo_root)\services\$($service)"
+
+    Get-ChildItem -Path $dir_path | ForEach-Object {
+        Write-Host ">>> Creating symbolic link for $($_.Name) ..."
+        $link_path = Join-Path -Path $target_dir -ChildPath $_.Name
+        try {
+            New-Item -ItemType SymbolicLink -Path $link_path -Target $_.FullName -ErrorAction Stop  | Out-Null
+        }
+        catch [System.IO.IOException] {
+            Write-Warning ">>> Symbolic link for $($_.Name) already exists."
+        }
+        catch {
+            Write-Warning ">>> Error creating symbolic link for $($_.Name): $_"
+        }
+    }
+    #>
+
+    $envFile = "$($repo_root)\services\$($service)\.env"
+    if (Test-Path $envFile -PathType Leaf) {
+        Write-Host ">>> Using $envFile as environment file."
+    } else {
+        Write-Warning ">>> No environment file found, you might be missing AYON_API_KEY..."
+    }
+    
+    Write-Host "--- running image $image"
```

Comment:
I don't know if windows allows symlinks or not, but the idea is that when you run the service, you end up with the following:
```
	processor/__init__.py
	processor/__pycache__
	processor/ayon_shotgrid_hub
	processor/constants.py
	processor/utils.py
```

which are the contents of the `services\shotgrid_common` into the `services\processor`, and then once the service is cancelled, they should be removed, is that feasible?

Reply:
Oh, I didn't see the `find $(CURDIR)/$(SERVICE) -type l -delete`. You can create symlinks on Windows (you need to have administrative privilegues for some reason) but with the docker command, you are running the image that already has those inside and then mounting volume with them again:

```
-v $(CURDIR)/shotgrid_common:$(CURDIR)/shotgrid_common:Z \
```
so those files doesn't have to be in `/services/processor`, or am I missing something?

(also that `:Z` in the volume argument doesn't work with vanilla docker, not sure what it should do?)

Reply:
Since they are symlinks to `/shotgrid_common`, unless i mount where they point at, the container wont be able to find them, they'd show up as broken links.

You can ignore teh `:z` is al inux only thing, for `SELinux`